src/main/java/com/smartcar/sdk/AuthClient.java

```diff
@@ -77,11 +80,30 @@ public Builder redirectUri(String redirectUri) {
       return this;
     }
 
+    /**
+     * @deprecated use {@link Builder#mode(String)} instead.
+     */
+    @Deprecated
     public Builder testMode(boolean testMode) {
-      this.testMode = testMode;
+      this.mode = testMode ? "test" : "live";
       return this;
     }
 
+    public Builder mode(String mode) {
+      final List<String> validModes = new ArrayList<String>(){
+        {add("test"); add("live");add("simulated");}
+      };
```

Comment:
```suggestion
      final Set<String> validModes = Stream.of("test", "live", "simulated").collect(Collectors.toSet());
```

Although this probably better to just have as a `private final` attribute on the `AuthClient` class so don't have to create the set everytime